File: scripts/create-assets.js
#!/usr/bin/env node
/**
 * Creates valid placeholder PNG assets for Expo.
 * Run: node scripts/create-assets.js
 */

const fs = require('fs');
const path = require('path');

// Minimal valid 1x1 rose-colored PNG (base64 encoded)
// This is a valid PNG that Expo can process
const VALID_PNG_BASE64 = 
  'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8' +
  'z8DwHwAFAAH/plvxnQAAAABJRU5ErkJggg==';

const assetsDir = path.join(__dirname, '..', 'assets');

// Ensure assets directory exists
if (!fs.existsSync(assetsDir)) {
  fs.mkdirSync(assetsDir, { recursive: true });
}

const files = ['icon.png', 'splash-icon.png', 'adaptive-icon.png', 'favicon.png'];
const pngBuffer = Buffer.from(VALID_PNG_BASE64, 'base64');

files.forEach(filename => {
  const filepath = path.join(assetsDir, filename);
  fs.writeFileSync(filepath, pngBuffer);
  console.log(`Created: ${filepath}`);
});

console.log('\nPlaceholder assets created!');
console.log('Replace with real icons (1024x1024 for icon/splash, 48x48 for favicon) before production.');

File: scripts/create-assets.py
#!/usr/bin/env python3
"""
Create placeholder PNG assets for Expo app.
Run this script from the project root:
    python3 scripts/create-assets.py
"""

import base64
import os

# Minimal 1x1 rose-colored PNG (placeholder)
PNG_DATA = base64.b64decode(
    'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP4z8DwHwAFAAH/plvxnQAAAABJRU5ErkJggg=='
)

def main():
    assets_dir = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'assets')
    os.makedirs(assets_dir, exist_ok=True)
    
    files = ['icon.png', 'splash-icon.png', 'adaptive-icon.png', 'favicon.png']
    
    for filename in files:
        filepath = os.path.join(assets_dir, filename)
        with open(filepath, 'wb') as f:
            f.write(PNG_DATA)
        print(f'Created: {filepath}')
    
    print('\nPlaceholder assets created! Replace with real icons before production.')

if __name__ == '__main__':
    main()

